syft/format/spdxjson/encoder.go

```diff
@@ -38,8 +39,9 @@ func NewFormatEncoderWithConfig(cfg EncoderConfig) (sbom.FormatEncoder, error) {
 
 func DefaultEncoderConfig() EncoderConfig {
 	return EncoderConfig{
-		Version: spdxutil.DefaultVersion,
-		Pretty:  false,
+		Version:           spdxutil.DefaultVersion,
+		Pretty:            false,
+		DeterministicUUID: false,
```

Comment:
We could make this the default as well. UUIDv4 and v5 are equally permitted by [the spec](https://spdx.github.io/spdx-spec/v2.3/document-creation-information/#652-intent)

Reply:
TY @dsseng for the contributon and sorry for the delay in getting back to this PR. While both are permitted by the spec I think for now we're going to allow this to be opt in behavior